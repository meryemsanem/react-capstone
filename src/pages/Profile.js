import React from 'react';
import { useSelector } from 'react-redux';
import './Profile.css';

function Profile() {
  const { missions, joinedMissions } = useSelector((store) => store.missions);
  const { data, joinedRockets } = useSelector((store) => store.rockets);
  const reservedMissions = missions.filter((mission) => joinedMissions.includes(mission.id));
  const reservedRockets = data.filter((rocket) => joinedRockets.includes(rocket.id));
  return (
    <div className="profile">
      <table className="profile-missions">
        <thead>
          <tr>
            <th>Missions Joined</th>
          </tr>
        </thead>
        <tbody>
          {reservedMissions.map((mission) => (
            <tr key={mission.id}>
              <td>
                {mission.name}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
      <table className="profile-rockets">
        <thead>
          <tr>
            <th>Rockets Reserved</th>
          </tr>
        </thead>
        <tbody>
          {reservedRockets.map((rocket) => (
            <tr key={rocket.id}>
              <td>{rocket.name}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}

export default Profile;