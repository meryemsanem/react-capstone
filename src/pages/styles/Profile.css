.profile {
  color: #333;
  display: flex;
  align-items: flex-start;
  margin-top: 20px;
}

.profile thead tr th {
  font-size: 28px;
  text-align: left;
  vertical-align: top;
  height: 50px;
}

.profile-missions,
.profile-rockets {
  width: 100%;
  margin: 0 80px;
  border-collapse: collapse;
}

.profile-missions tbody tr td,
.profile-rockets tbody tr td {
  border: 1px solid #999;
  height: 40px;
  vertical-align: top;
  padding: 6px;
  font-size: 18px;
}
