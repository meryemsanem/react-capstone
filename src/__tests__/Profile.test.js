import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Profile from '../pages/Profile';

const mockStore = configureStore([]);

test('renders Profile correctly', () => {
  const store = mockStore({
    missions: {
      missions: [
        { id: 1, name: 'Mission 1' },
        { id: 2, name: 'Mission 2' },
      ],
      joinedMissions: [1],
    },
    rockets: {
      data: [
        { id: 1, name: 'Rocket 1' },
        { id: 2, name: 'Rocket 2' },
      ],
      joinedRockets: [2],
    },
  });

  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Profile />
      </MemoryRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
