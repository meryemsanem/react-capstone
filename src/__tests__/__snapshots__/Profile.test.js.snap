// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile correctly 1`] = `
<DocumentFragment>
  <div
    class="profile"
  >
    <table
      class="profile-missions"
    >
      <thead>
        <tr>
          <th>
            My Missions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Mission 1
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class="profile-rockets"
    >
      <thead>
        <tr>
          <th>
            My Rockets
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Rocket 2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
